resolver.go

```diff
@@ -13,7 +13,7 @@ import (
 )
 
 type Resolver struct {
-	sync.RWMutex
+	mx sync.Mutex
```

Comment:
Figured exporting the Lock/Unlock functions here was unintentional and a bad idea for a consumer to use.

Reply:
ah yes, it was just programmatic convenience!